extern/cfe/src/bmi_cfe.c

```diff
@@ -1271,25 +1257,26 @@ Bmi* register_bmi(Bmi *model) {
         model->set_value = Set_value;
         model->set_value_at_indices = Set_value_at_indices;
 
-        model->get_grid_size = Get_grid_size;    // TODO: needs finished implementation
-        model->get_grid_rank = Get_grid_rank;    // TODO: needs finished implementation
-        model->get_grid_type = Get_grid_type;    // TODO: needs finished implementation
-
-        model->get_grid_shape = Get_grid_shape;    // TODO: needs finished implementation
-        model->get_grid_spacing = Get_grid_spacing;    // TODO: needs finished implementation
-        model->get_grid_origin = Get_grid_origin;    // TODO: needs finished implementation
-
-        model->get_grid_x = Get_grid_x;    // TODO: needs finished implementation
-        model->get_grid_y = Get_grid_y;    // TODO: needs finished implementation
-        model->get_grid_z = Get_grid_z;    // TODO: needs finished implementation
-
-        model->get_grid_node_count = Get_grid_node_count;    // TODO: needs finished implementation
-        model->get_grid_edge_count = Get_grid_edge_count;    // TODO: needs finished implementation
-        model->get_grid_face_count = Get_grid_face_count;    // TODO: needs finished implementation
-        model->get_grid_edge_nodes = Get_grid_edge_nodes;    // TODO: needs finished implementation
-        model->get_grid_face_edges = Get_grid_face_edges;    // TODO: needs finished implementation
-        model->get_grid_face_nodes = Get_grid_face_nodes;    // TODO: needs finished implementation
-        model->get_grid_nodes_per_face = Get_grid_nodes_per_face;    // TODO: needs finished implementation
+        model->get_grid_size = Get_grid_size;    
+        model->get_grid_rank = Get_grid_rank;    
+        model->get_grid_type = Get_grid_type;    
+
+        model->get_grid_shape = NULL;
+        model->get_grid_spacing = NULL;
+        model->get_grid_origin = NULL;
```

Comment:
What motivated removing function pointer assignments to the "empty" function implementations - e.g., `model->get_grid_x = Get_grid_x` - and instead setting the pointer to `NULL`?  Off the top of my head, I don't recall what would happen if something tried to call a function using a null pointer like this, but I would (i.e. did) think it makes more sense to have the empty implementation that just immediately returns the error code.  Is there something in the spec suggesting this is a more correct approach?

Reply:
I was simply following [csdms BMI C example](https://github.com/csdms/bmi-example-c/blob/master/heat/bmi_heat.c).  Most of the model grid functions aren't applicable because we defined grid type to be of "scalar".  
Now looking at [BMI best practices](https://bmi.readthedocs.io/en/latest/bmi.best_practices.html#best-practices), which states that all BMI functions must be implemented, even if empty.
Best to "do as I say, not what I do"?  I'll go ahead revert back to the original setup for these, using the empty functions assignments. -thanks

Reply:
Yeah, better than the `segfault` that would likey occur and drive someone crazy to figure out why...